feat: Add theme mode selection

notJoon opened the pull request:
# Description

- Add theme mode selection (System/Light/Dark) with a dropdown menu
- consistent theme across all UI components

Related with #9 

## Example

**Light Mode**

<img width="1220" alt="Screenshot 2025-04-28 at 6 59 04 PM" src="https://github.com/user-attachments/assets/98be63c1-8a30-4e3f-ba30-48f2b2311dc4" />

**Dark Mode**

<img width="1220" alt="Screenshot 2025-04-28 at 6 59 48 PM" src="https://github.com/user-attachments/assets/d769e469-e9b4-4155-a7c0-6758ae823777" />

faiface reviewed (changes requested):
Wonderful!

There is one problem, though. The System theme doesn't work right.

When I switch to Light, and then to System, it stays light. If I switch to Dark, and then System, it stays dark.

In other words, System seems to use the last selected theme.

notJoon commented:
@faiface Thank you for your feedback. I have fixed the issue in this [commit](https://github.com/faiface/par-lang/pull/42/commits/f9703242df969fbcc31853dbb2d94ca57227d843). Could you please check it? I've added a demo video to the description for your reference. Thank you 👍

faiface reviewed (changes requested):
Works great! Nice video too ;)

One more small thing:

<img width="426" alt="Untitled" src="https://github.com/user-attachments/assets/0cd03bb1-70e4-4ffb-90b1-3cc18030f155" />

Somehow we got in these grey backgrounds on some labels, not sure how that happened...

Before your change, it was like this:

<img width="426" alt="Screenshot 2025-04-28 at 14 54 10" src="https://github.com/user-attachments/assets/9a63c76d-f49a-4d44-95f0-d90559df3153" />

Could you try and fix this? After that, it's perfect.

notJoon commented:
> Could you try and fix this? After that, it's perfect.

After checking, I found that `set_visual` was being called on every frame, which was overriding all styles with default values. So the custom styles were also being reset, causing the background color to appear unintentionally. I've modified it to reapply the custom styles after each `set_visuals` call on every frame.

<img width="1112" alt="Screenshot 2025-04-28 at 10 50 51 PM" src="https://github.com/user-attachments/assets/86f697b2-e794-41cf-aa2c-98de386c6143" />

After the modification, it appears that the issue has been resolved.

notJoon commented:
> Working with UI is fun, isn't it? 😄
@faiface 
Yep The UI is certainly quite fun (honestly, I didn't expect it to be this difficult 😅). I applied your suggestions and did more testing to confirmed the following points:

- Background color remains transparent for all labels except selectable buttons
- Even when the system theme changes, the theme explicitly selected by the user is maintained

Below is the demo

![Screen Recording 2025-04-29 at 10 36 10 AM](https://github.com/user-attachments/assets/32ddc896-7bf1-4d43-9aa1-b9ab21912cbd)

Also, `cargo fmt` too
